core/modules/xrdsvc/SsiProvider.cc

```diff
@@ -0,0 +1,148 @@
+// -*- LSST-C++ -*-
+/*
+ * LSST Data Management System
+ * Copyright 2014-2015 AURA/LSST.
+ *
+ * This product includes software developed by the
+ * LSST Project (http://www.lsst.org/).
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the LSST License Statement and
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+/// Implement XrdSsiProviderServer to provide Qserv's SsiService
+/// implementation. Link this file when building a plugin to be used as
+/// ssi.svclib or oss.statlib.
+
+// Class header
+#include "xrdsvc/SsiProvider.h"
+
+// System headers
+#include <memory>
+#include <sstream>
+#include <sys/types.h>
+
+// Third party headers
+#include "XrdSsi/XrdSsiCluster.hh"
+#include "XrdSsi/XrdSsiLogger.hh"
+
+// LSST headers
+#include "lsst/log/Log.h"
+
+// Qserv headers
+#include "global/ResourceUnit.h"
+#include "wpublish/ChunkInventory.h"
+#include "xrdsvc/XrdName.h"
+
+/******************************************************************************/
+/*                               G l o b a l s                                */
+/******************************************************************************/
+
+XrdSsiProvider *XrdSsiProviderServer = 
+                new  lsst::qserv::xrdsvc::SsiProviderServer;
+
+XrdSsiProvider *XrdSsiProviderLookup = XrdSsiProviderServer;
+  
+/******************************************************************************/
+/*                                  I n i t                                   */
+/******************************************************************************/
+  
+namespace lsst {
+namespace qserv {
+namespace xrdsvc {
+
+bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
+          char const*   parms, int argc, char **argv) {
+
+    lsst::qserv::xrdsvc::XrdName x;
+
+// Establish our instance name (many different qservs may be running here).
+//
```

Comment:
Ah, but that still violates the rules in my view because the whole point 
here is that code is not indented relative to the comment. In my view, the 
comment is an "if" and the code that follows is the subject of the 
contextual "if" which makes you naturally indent code relative to the 
comment. That is not allowed; making comments take on the same importance 
as the code. That's rarely the case for no other reason is that the 
comment may be outdated. Comments should be treated as subjective 
statements on what is actually happening. Hence, the code should take on 
more importance. The rules don't recognize that. I'm OK with it but I 
think it's not very realistic.

Andy

On Wed, 14 Oct 2015, Andy wrote:

>> +
>> +/******************************************************************************/
>> +/*                                  I n i t                                   */
>> +/******************************************************************************/
>> +
>> +namespace lsst {
>> +namespace qserv {
>> +namespace xrdsvc {
>> +
>> +bool SsiProviderServer::Init(XrdSsiLogger* logP,  XrdSsiCluster* clsP, char const* cfgFn,
>> +          char const*   parms, int argc, char **argv) {
>> +
>> +    lsst::qserv::xrdsvc::XrdName x;
>> +
>> +// Establish our instance name (many different qservs may be running here).
>> +//
>
> Yes, spaces make code more readable. IT's not really a mandatory 4-space indent, I think namespace-level code has no indent, so if you comment code at namespace level there will be no indent. Inside normal methods/classes it is indeed multiples of 4 spaces.
>
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/lsst/qserv/pull/143/files#r42085608